ipaserver/install/certs.py

```diff
@@ -501,13 +501,19 @@ def create_password_conf(self):
         This is the format of mod_nss pin files.
         """
         ipautil.backup_file(self.pwd_conf)
-        f = open(self.pwd_conf, "w")
-        f.write("internal:")
-        pwdfile = open(self.passwd_fname)
-        f.write(pwdfile.read())
-        f.close()
-        pwdfile.close()
-        self.set_perms(self.pwd_conf, uid=constants.HTTPD_USER)
+
+        with open(self.passwd_fname) as pwdfile:
+            password = pwdfile.read()
+
+        with open(self.pwd_conf, "w") as f:
+            f.write("internal:")
+            f.write(password)
+            f.write("\nNSS FIPS 140-2 Certificate DB:")
+            f.write(password)
+            # make sure other processes can access the file contents ASAP
+            f.flush()
+        if os.path.exists(self.pwd_conf):
```

Comment:
Why do you need to check presence of file that is actually created a few lines above?